src/ManagerV2/FolderableEntityManager.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\ManagerV2;
+
+use App\Entity\Document;
+use App\Entity\Dossier;
+use App\Entity\Interface\FolderableEntityInterface;
+use Doctrine\ORM\EntityManagerInterface;
+
+class FolderableEntityManager
+{
+    public function __construct(
+        private readonly EntityManagerInterface $em,
+    ) {
+    }
+
+    public function move(FolderableEntityInterface $folderableEntity, ?Dossier $folder): void
+    {
+        if ($folderableEntity instanceof Dossier) {
+            $this->moveFolder($folderableEntity, $folder);
```

Comment:
This is strange, your interface should enforce some methods to avoid this strange thing